public/app/plugins/datasource/prometheus/language_provider.ts

```diff
@@ -443,23 +443,27 @@ export default class PromQlLanguageProvider extends LanguageProvider {
    */
   fetchSeriesLabels = async (name: string, withName?: boolean): Promise<Record<string, string[]>> => {
     const tRange = this.datasource.getTimeRange();
-    const params = new URLSearchParams({
+    const urlParams = new URLSearchParams({
```

Comment:
Shouldn't this be a plain object, not `URLSearchParams`?

Reply:
It seems like it should :D I'm getting an error if it's not. Could this be unit tested?

Reply:
I left URLSearchParams on purpose, because then we can easily create cache key by using`toString()` method `cacheParams.toString()}`. If we used plain object, we would have to do `Object.entries(data).map(([k, v]) => `${encodeURIComponent(k)}=${encodeURIComponent(v)}`).join('&');`.

Reply:
We can also do that if you think it is more consistent and less confusing. I am honestly not sure which would be better.

Reply:
For `cacheParams` it's absolutely fine, but `urlParams` are passed to `this.request(url, [], urlParams);` down the line. URLSearchParams doesn't work like a map and params are not added to the URL in `this._request`.

Reply:
Good catch! Updated in https://github.com/grafana/grafana/pull/33321/commits/c96a33f46d2f2f24bbb09a22be3c1b56ce467781.